src/components/EntityScorecardsCard.tsx

```diff
@@ -0,0 +1,232 @@
+import React, { useState } from "react";
+import { InfoCard } from "@backstage/core-components";
+import { useEntity } from "@backstage/plugin-catalog-react";
+import useAsync from "react-use/lib/useAsync";
+import { useApi } from "@backstage/core-plugin-api";
+import { Progress, ResponseErrorPanel } from "@backstage/core-components";
+import Box from "@material-ui/core/Box";
+import Tabs from "@material-ui/core/Tabs";
+import Tab from "@material-ui/core/Tab";
+
+import { dxApiRef, Scorecard, ScorecardCheck } from "../api";
+import { BrandedCardTitle } from "./BrandedCardTitle";
+import { CheckResultBadge } from "./CheckResultBadge";
+import { COLORS } from "../styles";
+
+const DEFAULT_NO_LEVEL_COLOR = "#CBD5E1";
+
+type EntityScorecardsCardProps = {
+  contentMaxHeight?: string | number;
+};
+
+export function EntityScorecardsCard({
+  contentMaxHeight = "20rem",
+}: EntityScorecardsCardProps) {
+  const dxApi = useApi(dxApiRef);
+
+  const { entity } = useEntity();
+
+  const entityIdentifier = entity.metadata.name;
+
+  const [tab, setTab] = useState<"levels" | "checks">("levels");
+
+  const {
+    value: response,
+    loading,
+    error,
+  } = useAsync(() => {
+    return dxApi.scorecards(entityIdentifier);
+  }, [dxApi, entityIdentifier]);
+
+  if (loading) {
+    return <Progress />;
+  }
+
+  if (error) {
+    return <ResponseErrorPanel error={error} />;
+  }
+
+  if (!response) {
+    throw new Error("Unreachable");
+  }
+
+  const scorecards = response.scorecards;
+
+  const flattenedChecks = scorecards.flatMap((scorecard) => scorecard.checks);
+
+  return (
+    <InfoCard
+      title={
+        <Box
+          sx={{
+            display: "flex",
+            flexDirection: "column",
+            gridGap: 16,
+            marginBottom: -20,
+          }}
+        >
+          <BrandedCardTitle title="Scorecards" />
+          <Tabs
+            value={tab}
+            onChange={(_, value) => setTab(value)}
+            indicatorColor="primary"
+          >
+            <Tab value="levels" label="Levels" />
+            <Tab value="checks" label="Checks" />
+          </Tabs>
+        </Box>
+      }
+      deepLink={{
+        link: `https://app.getdx.com/catalog/${entityIdentifier}/scorecards`,
```

Comment:
I wonder if this should be something we grab from our own API's. That way we pass the backstage string entity ref to our API and return the entity identifier DX has. This covers the case when they have a different identifier in DX through an Alias.